src/ompl/base/samplers/src/DeterministicStateSampler.cpp

```diff
@@ -0,0 +1,150 @@
+*     notice, this list of conditions and the following disclaimer.
+*   * Redistributions in binary form must reproduce the above
+*     copyright notice, this list of conditions and the following
+*     disclaimer in the documentation and/or other materials provided
+*     with the distribution.
+*   * Neither the name of the Rice University nor the names of its
+*     contributors may be used to endorse or promote products derived
+*     from this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+/* Author: Leonard Bruns */
+
+#include "ompl/base/samplers/DeterministicStateSampler.h"
+#include "ompl/base/samplers/deterministic/HaltonSequence.h"
+#include "ompl/base/spaces/SE2StateSpace.h"
+#include "ompl/base/spaces/SO2StateSpace.h"
+#include "ompl/base/spaces/RealVectorStateSpace.h"
+
+#include <iostream>
+#include <boost/math/constants/constants.hpp>
+
+namespace ompl {
+    namespace base {
+        DeterministicStateSampler::DeterministicStateSampler(const StateSpace *space, DeterministicSamplerType type) :
+            StateSampler(space) {
+            switch(type) {
+                case HALTON:
+                    sequence_ptr_ = std::make_shared<HaltonSequence>(space->getDimension());
+                    break;
+                default:
+                    std::cout << "Warning: Unknown deterministic sampler type specified, using Halton instead." << std::endl;
+                    break;
+            }
+        }
+
+        DeterministicStateSampler::DeterministicStateSampler(const StateSpace *space, std::shared_ptr<DeterministicSequence> sequence_ptr) :
+            StateSampler(space), sequence_ptr_(sequence_ptr) {
+
+        }
+
+        void SO2DeterministicStateSampler::sampleUniform(State *state) {
+            auto sample = sequence_ptr_->sample();
+            state->as<SO2StateSpace::StateType>()->value = -boost::math::constants::pi<double>() + sample[0]*2*boost::math::constants::pi<double>();
+        }
+
+        void SO2DeterministicStateSampler::sampleUniformNear(State *state, const State *near, double distance) {
+            (void) state;
```

Comment:
I'm guessing you're doing this to suppress compiler warnings. Don't do it this way. Either omit the argument names or comment them out C-style:
```
 SO2DeterministicStateSampler::sampleUniformNear(State *, const State *, double) 
```
or
```
 SO2DeterministicStateSampler::sampleUniformNear(State */*state*/, const State */*near*/, double /*distance*/) 
```

Reply:
done